homeassistant/components/ecoforest/number.py

```diff
@@ -0,0 +1,77 @@
+"""Support for Ecoforest number platform."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pyecoforest.models.device import Device
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import EcoforestCoordinator
+from .entity import EcoforestEntity
+
+
+@dataclass
+class EcoforestRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[Device], float | None]
+
+
+@dataclass
+class EcoforestNumberEntityDescription(
+    NumberEntityDescription, EcoforestRequiredKeysMixin
+):
+    """Describes an ecoforest number entity."""
+
+
+NUMBER_ENTITIES = (
+    EcoforestNumberEntityDescription(
+        key="power",
+        translation_key="power",
+        name="Power",
```

Comment:
The power identifies the target power of the device, it can be set from 1 to 9.

Reply:
Is power also used as name for the setting in the app? I am thinking it might get people confused with power as in kW. (As the name will be translated, I think it might get translated to the power in kW variant in some languages, so I'm trying to avoid that)

Reply:
I understand it might be difficult to have the full picture on how the device operates, would be good to include in the "standards" a markdown file with device information and capabilities, I believe this would be good for reviewers to have the big picture of the device.

That being said, I also spent some time thinking on what would be the best approach and namings for the device.
This device is a fireplace that have the following capabilities:
* Operation mode: On of the following modes are valid: Temperature or Power/Potency
* Target: When in temperature mode the target could be a temperature and when in potency mode it requires to be a integer between 1 and 9. In the API/CGI of the device this is set by 2 different parameters (in spanish consigna_potencia and consigna_temperatura).
* on and off switch: the allow to set the device on and off, while the device is starting or shutting dow it will iterate over a state machine, it doesn't go directly to on and off state. but this parameter in the API/CGI allow us to set on and off. The switch is set to off only when the state is off otherwise it's on since the device is working. Ideally while shutting down and powering on the state shouldn't be changeable either.

After this what are your thoughts?

Reply:
How does it switch between temperature mode and power mode? Does it automatically switch when you update one of the two? Or will you add a select entity for that in the future?

Maybe "power level" covers it.

Reply:
It doesn't switch automatically but it can be done in the API/CGI so in the future a select can be added. For now only power will be supported. 

Power level seems fine to me, also considered it, also though on power target and temperature target.

Reply:
I'm leaving that for you (although I do want to mention target temperature sounds better ;) )